src/bindgen/cdecl.rs

```diff
@@ -178,11 +186,17 @@ impl CDecl {
             let next_is_pointer = iter_rev.peek().map_or(false, |x| x.is_ptr());
 
             match *declarator {
-                CDeclarator::Ptr(ref is_const) => {
-                    if *is_const {
-                        out.write("*const ");
+                CDeclarator::Ptr(ref is_const, ref is_nonnull) => {
```

Comment:
Just keep the code like it was, then do:

```
if is_nonnull {
    if let Some(ref attr) = config.non_null_attribute {
        write!(out, "{} ", attr);
    }
}
```

? I don't see why having the match on both is useful.

Reply:
Done.